libtransact/src/context/manager.rs

```diff
@@ -0,0 +1,528 @@
+                                contexts.push_back(self.get_context(context_id)?);
+                            }
+                        }
+                    }
+                }
+                // Once the Context which contains the Key is found, it is used to access the value
+                if context.contains(&key) {
+                    let mut_context = self.get_context_mut(context.id())?;
+                    if let (_, Some(deleted_value)) = mut_context.delete_state(key) {
+                        return_state_change = Some(StateChangeResult::Delete {
+                            value: deleted_value,
+                        });
+                    }
+                } else if let Some(value) = self
+                    .database
+                    .get(context.state_id(), &[key.clone()])
+                    .expect("Unable to get State")
+                    .get(&key)
+                {
+                    self.get_context_mut(&context_id)?.delete_state(key);
+                    return_state_change = Some(StateChangeResult::Delete {
+                        value: value.clone(),
+                    });
+                }
+                Ok(return_state_change)
+            }
+        }
+    }
+
+    /// Adds an Event to the specified Context.
+    pub fn add_event(
+        &mut self,
+        context_id: ContextId,
+        event: Event,
+    ) -> Result<(), ContextManagerError> {
+        let context = self.get_context_mut(&context_id)?;
+        context.add_event(event);
+        Ok(())
+    }
+
+    /// Adds Data to the specified Context.
+    pub fn add_data(
+        &mut self,
+        context_id: ContextId,
+        data: Vec<u8>,
+    ) -> Result<(), ContextManagerError> {
+        let context = self.get_context_mut(&context_id)?;
+        context.add_data(data);
+        Ok(())
+    }
+
+    /// Creates a Context, and returns the resulting ContextId.
+    pub fn create_context(
+        &mut self,
+        dependent_contexts: Vec<ContextId>,
+        state_id: &str,
+    ) -> ContextId {
+        let new_context = Context::new(state_id, dependent_contexts);
+        self.contexts.insert(*new_context.id(), new_context.clone());
+        *new_context.id()
+    }
+
+    /// Creates a TransactionReceipt based on the information available within the specified Context.
+    pub fn get_transaction_receipt(
+        &self,
+        context_id: ContextId,
+        transaction_id: &str,
+    ) -> Result<TransactionReceipt<K, V>, ContextManagerError> {
+        let context = self.get_context(&context_id)?;
+        let new_transaction_receipt = TransactionReceiptBuilder::new()
+            .with_state_changes(context.state_changes().to_vec())
+            .with_events(context.events().to_vec())
+            .with_data(context.data().to_vec())
+            .with_transaction_id(transaction_id.to_string())
+            .build()
+            .unwrap();
```

Comment:
Should convert the build error here to a ContextManagerError.

Reply:
Addressed